Fix #970, sort feature types

mfeurer opened the pull request:
Closes #970

When encoding a pandas array in autosklearn.data.validator, the columns are re-ordered by the ColumnTransformer. This PR re-orders the feature types so that when passing the data to the actual ML pipeline, columns and feature types are sorted the same way.

franchuterivera commented:
I would have expected this check to catch this:

https://github.com/automl/auto-sklearn/blob/50e94326a61aaa7b1a946ba938e2d70a3fa9ac51/test/test_automl/test_estimators.py#L493

Maybe it makes sense to consider having 1558 there (dataset 2 did not seem to catch it ) and covertype categories seem pretty decent? 
```
V1      float64
V2     category
V3     category
V4     category
V5     category
V6      float64
V7     category
V8     category
V9     category
V10     float64
V11    category
V12     float64
V13     float64
V14     float64
V15     float64
V16    category
dtype: object

```

mfeurer commented:
That's a good suggestion, however, dataset 1558 doesn't have missing values. I just checked OpenML for datasets with nominal values, missing values, negative real-values, classification as target and being small, but didn't find any.

Therefore, I suggest keeping the test case as it is as we have fixed this issue and added a test at the correct part in the code.
